test/integration/login_test.rb

```diff
@@ -0,0 +1,39 @@
+require 'test_helper'
+
+class LoginTest < ActionDispatch::IntegrationTest
+  setup do
+    Capybara.current_driver = Capybara.javascript_driver # :selenium by default
+    visit '/users/sign_in'
+  end
+
+  test 'shows login initially' do
+    assert has_current_path?('/users/sign_in')
+  end
+
+  test 'create an account, sign out and log in again' do
+    new_email = 'some.email@example.com'
+    new_password = 'password'
+    click_on 'Sign up'
+    assert has_current_path?('/users/sign_up')
+    
+    fill_in 'new_email', with: new_email
+    fill_in 'new_password', with: new_password
+    fill_in 'confirm_password', with: new_password
+    click_on 'Sign up'
+    assert has_current_path?('/')
+    assert page.has_content?("Twitter Clone")
+    
+    # find('.likes-and-replies--reaction-counter-inner', match: first)
+    sleep(inspection_time=0.5)
```

Comment:
Yeah I found out that the issue was that once I logged in, a bunch of fetch requests were called to get the posts + replies + count. It loads super fast but when capybara log the user out immediately it flashes a failed to fetch modal since the user isn't authenticated by the time the fetch gets to it.

I tried finding a wait until method or something similar online and spacewaffle and it seems everyone just creates a `areAjaxRequestsComplete? else loop sleep(0.5)` including spacewaffle which Brandon showed me.